### PR TITLE
Add Canonicalization Patterns to Fold a BatchNorm into Conv wts in the BN(Add(Conv, V)) sequence

### PR DESCRIPTION
**Pull request by jorickert**

### CODE REVIEW

File: test/mlir/onnx/onnx_canonicalization.mlir [hunk 1]
```diff
@@ -1243,6 +1290,487 @@ func.func @test_no_fuse_mul_conv_batch_scaling(%arg0: tensor<1x1x28x28xf32>) ->
...
+
+    // CHECK-LABEL:  func.func @test_distribute_mul_over_add_one_const
+    // CHECK-SAME:   ([[PARAM_0_:%.+]]: tensor<1x16x8x8xf32>) -> tensor<1x16x8x8xf32> {
+    // CHECK-DAG:       [[VAR_0_:%.+]] = onnx.Constant dense<1.000000e+00> : tensor<1x16x8x8xf32>
+    // CHECK-DAG:       [[VAR_1_:%.+]] = onnx.Constant dense<2.000000e+00> : tensor<1x16x1x1xf32>
+    // CHECK-DAG:       [[VAR_2_:%.+]] = "onnx.Mul"([[PARAM_0_]], [[VAR_1_]]) : (tensor<1x16x8x8xf32>, tensor<1x16x1x1xf32>) -> tensor<1x16x8x8xf32>
+    // CHECK-DAG:       [[VAR_3_:%.+]] = "onnx.Mul"([[VAR_0_]], [[VAR_1_]]) : (tensor<1x16x8x8xf32>, tensor<1x16x1x1xf32>) -> tensor<1x16x8x8xf32>
+    // CHECK:           [[VAR_4_:%.+]] = "onnx.Add"([[VAR_2_]], [[VAR_3_]]) : (tensor<1x16x8x8xf32>, tensor<1x16x8x8xf32>) -> tensor<1x16x8x8xf32>
+    // CHECK:           return [[VAR_4_]] : tensor<1x16x8x8xf32>
+}
```
Comment (ge28boj) [inline test/mlir/onnx/onnx_canonicalization.mlir]: In this case before/after we'll have one Mul and one Add, but in different ordering. Does it mean Mul -> Add is preferred? Why is that?

Comment (jorickert) [reply test/mlir/onnx/onnx_canonicalization.mlir]: I somehow thought that this could be completely folded.
I think that switching add and mul is general beneficial, as it can for example be matched to a Batch Norm kernel or FMA , but i removed it from this PR for now to reduce the risk of accidental negative impacts

**Comment by jorickert** [comment]
- [ ] measure performance
